scipy/interpolate/tests/test_rgi.py

```diff
@@ -621,93 +687,139 @@ def test_xi_nd(self):
+                     method=method, bounds_error=False)
+        assert_allclose(v1, v2.reshape(v1.shape))
 
-    def test_nonscalar_values(self):
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values(self, method):
         # Verify that non-scalar valued values also works
-        points, values = self._sample_4d_data()
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5)] * 2 + [
+            (0.0, 5.0, 10.0, 15.0, 20, 25.0)
+        ] * 2
 
-        np.random.seed(1234)
-        values = np.random.rand(3, 3, 3, 3, 6)
-        sample = np.random.rand(7, 11, 4)
+        rng = np.random.default_rng(1234)
+        values = rng.random((6, 6, 6, 6, 8))
+        sample = rng.random((7, 11, 4))
 
-        for method in ['nearest', 'linear']:
-            v = interpn(points, values, sample, method=method,
-                        bounds_error=False)
-            assert_equal(v.shape, (7, 11, 6), err_msg=method)
+        v = interpn(points, values, sample, method=method,
+                    bounds_error=False)
+        assert_equal(v.shape, (7, 11, 8), err_msg=method)
 
-            vs = [interpn(points, values[...,j], sample, method=method,
-                          bounds_error=False)
-                  for j in range(6)]
-            v2 = np.array(vs).transpose(1, 2, 0)
+        vs = [interpn(points, values[..., j], sample, method=method,
+                      bounds_error=False) for j in range(8)]
+        v2 = np.array(vs).transpose(1, 2, 0)
 
-            assert_allclose(v, v2, err_msg=method)
+        assert_allclose(v, v2, atol=1e-14, err_msg=method)
 
+    @parametrize_rgi_interp_methods
+    def test_nonscalar_values_2(self, method):
+        # Verify that non-scalar valued values also work : use different
+        # lengths of axes to simplify tracing the internals
+        points = [(0.0, 0.5, 1.0, 1.5, 2.0, 2.5),
+                  (0.0, 0.5, 1.0, 1.5, 2.0, 2.5, 3.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0),
+                  (0.0, 5.0, 10.0, 15.0, 20, 25.0, 35.0, 36.0, 47)]
+
+        rng = np.random.default_rng(1234)
+
+        num_trailing_dims = 2
+        trailing_points = [3 + i for i in range(num_trailing_dims)]
+        # NB: values has a `num_trailing_dims` trailing dimension
+        values = rng.random((6, 7, 8, 9, *trailing_points))
+        sample = rng.random(4)   # a single sample point !
+
+        v = interpn(points, values, sample, method=method, bounds_error=False)
+
+        # v has a single sample point *per entry in the trailing dimensions*
+        assert v.shape == (1, *trailing_points)
+
+        # check the values, too : manually loop over the trailing dimensions
+        vs = [[
+                interpn(points, values[..., i, j], sample, method=method,
+                        bounds_error=False) for i in range(values.shape[-2])
+              ] for j in range(values.shape[-1])]
+
+        assert_allclose(v, np.asarray(vs).T, atol=1e-14, err_msg=method)
```

Comment:
`.T` is a bit confusing(-ly different from `.transpose(1, 2, 0)` in the previous test)